lua/entities/kleiner/sv_spawning.lua

```diff
@@ -0,0 +1,62 @@
+-- This file is subject to copyright - contact swampservers@gmail.com for more information.
+-- INSTALL: CINEMA
+
+KLEINER_NPCS = KLEINER_NPCS or {}
+KLEINER_NPCS_FILTER = KLEINER_NPCS_FILTER or {}
+KLEINER_NPC_TARGETS = KLEINER_NPC_TARGETS or {}
+KLEINER_NPCS_CURRENT_NUMBER = KLEINER_NPCS_CURRENT_NUMBER or 0 
+KLEINER_DESIRED_NUMBER = 18
+
+
+
+
+	
+timer.Create("kleiner_spawner",1,0,function()
+	if(table.Count(KLEINER_NPCS) < KLEINER_DESIRED_NUMBER)then
+		local newkleiner = ents.Create("kleiner")
+		if(IsValid(newkleiner))then
+			local spawnpoint = gmod.GetGamemode().SpawnPoints and table.Random(gmod.GetGamemode().SpawnPoints):GetPos() or Vector(0,0,16)
+			newkleiner:SetPos(spawnpoint)
+			newkleiner:Spawn()
+			newkleiner:Activate()
+		end
+	end 
+end)  
+
+hook.Add("EntityRemoved","kleiner_npc_kleanup",function(ent)
+	if(ent:GetClass() == "kleiner" or ent:IsPlayer())then
+		KLEINER_NPCS[ent] = nil
+		KLEINER_NPC_TARGETS[ent] = nil
+		KLEINER_NPCS_CURRENT_NUMBER = table.Count(KLEINER_NPCS)
+	end	
+	
+end)
+hook.Add("OnEntityCreated","kleiner_npc_register",function(ent)
+	if(ent:GetClass() == "kleiner" or ent:IsPlayer())then
+		KLEINER_NPCS[ent] = ent:GetClass() == "kleiner" and true or nil
+		KLEINER_NPC_TARGETS[ent] =  ent:IsPlayer() and true or nil
+		
+		KLEINER_NPCS_FILTER = {}
```

Comment:
Remove this, use a function for the trace filter instead (if you were going to use it)